techSpec.js

```diff
@@ -0,0 +1,209 @@
+/*
```

Comment:
style nit: if you were writing a technical spec on Github, you'd normally use markdown.

Reply:
in a README file?

Reply:
The latest commit is in js still for ease of review. Once the content is correct, I will move to a README file.